### PR TITLE
[AArch64] Add support for Cortex-A520AE and Cortex-A720AE CPUs

### PR DESCRIPTION
**Pull request by jthackray**
[AArch64] Add support for Cortex-A520AE and Cortex-A720AE CPUs

Cortex-A520AE and Cortex-A720AE are Armv9.2 AArch64 CPUs.

Technical Reference Manual for Cortex-A520AE:
   https://developer.arm.com/documentation/107726/latest/

Technical Reference Manual for Cortex-A720AE:
   https://developer.arm.com/documentation/102828/latest/

### CODE REVIEW

File: llvm/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -67,6 +67,8 @@ Changes to Interprocedural Optimizations
 Changes to the AArch64 Backend
 ------------------------------
 
+* Added support for Cortex-A520AE and Cortex-A720AE CPUs.
```
Comment (davemgreen) [inline llvm/docs/ReleaseNotes.rst:70]: Could this have Cortex-A78AE too?

File: llvm/lib/Target/AArch64/AArch64Subtarget.h [hunk 2]
```diff
@@ -58,6 +58,7 @@ class AArch64Subtarget final : public AArch64GenSubtargetInfo {
     CortexA55,
     CortexA510,
     CortexA520,
+    CortexA520AE,
```
Comment (davemgreen) [inline llvm/lib/Target/AArch64/AArch64Subtarget.h:61]: These might not be worth adding, considering they should be the same as CortexA520, and could reuse the same enum.

File: llvm/docs/ReleaseNotes.rst (hunk 1 above)
Comment (jthackray) [reply llvm/docs/ReleaseNotes.rst]: Sure, now fixed.

File: llvm/lib/Target/AArch64/AArch64Subtarget.h (hunk 2 above)
Comment (jthackray) [reply llvm/lib/Target/AArch64/AArch64Subtarget.h]: Yes, good idea. Amended.

**Review by davemgreen — APPROVED** [review]
Thanks. LGTM

**Pull request merged**